### PR TITLE
[AIRFLOW-3029] New Operator - SqlOperator

### PR DESCRIPTION
**Pull request by jbacon**
Introduces a new `SqlOperator` that executes sql code in a database. Surprisingly this operator doesn't already exist. It is not always necessary to implement individual operators for each SQL database (`PostgresOperator`, `MySqlOperator`,etc..), which this operator shows. Similar to `SqlSensor`.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3029) issues and references them in the PR title. 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This abstract operator can be instantiated directly,
and does not need to be derived into subclasses for each DbApiHook subclass.
It will automatically uses the correct DbApiHook subclass implementation,
made possible by reflecting upon the specified Connection's `conn_type`.

Date:      Fri Sep 7 16:06:56 2018 -0700
new file:   airflow/contrib/operators/sql_operator.py
new file:   tests/contrib/operators/test_sql_operator.py

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

`/tests/contrib/operators/test_sql_operator.py`

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by Fokko** [comment]
This looks quite similar to the: https://github.com/apache/incubator-airflow/blob/master/airflow/operators/generic_transfer.py

**Comment by jbacon** [comment]
The `SqlOperator` has a slightly different use-case than the `GenericTransfer` operator, but they both similarly abstract the `DbApiHook` from the operator via `BaseHook.get_hook(..)`, hence why we don't have a `PostgresGenericTransfer`, `MySqlGenericTransfer`, or etc..  `SqlOperator` isn't intended to fetch any rows from the database, just execute SQL with either success/failure.

**Comment by jbacon** [comment]
This would be my first Apache commit, would anyone care to review it for me? is there anything else I can do to get it merged before next release.

File: tests/contrib/operators/test_sql_operator.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        ]
+        task_multi = SqlOperator(task_id='sql_operator_postgres_test_multi',
+                                 sql=sql,
+                                 conn_id='postgres_default',
+                                 dag=self.dag)
+        task_multi.run(start_date=self.DEFAULT_DATE,
+                       end_date=self.DEFAULT_DATE,
+                       ignore_ti_state=True)
+
+    @skipUnlessImported('airflow.operators.mysql_operator', 'MySqlOperator')
```
Comment (Fokko) [inline tests/contrib/operators/test_sql_operator.py:72]: Can you fix the tests by removing the `skipUnlessImported`? I don't think they are being executed right now.

File: tests/contrib/operators/test_sql_operator.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    DEFAULT_DATE = timezone.datetime(2015, 1, 1)
+
+    def setUp(self):
+        configuration.load_test_config()
+        args = {'owner': 'airflow', 'start_date': self.DEFAULT_DATE}
+        dag = DAG('unit_test_dag', default_args=args)
+        self.dag = dag
+
+    @skipUnlessImported('airflow.operators.postgres_operator', 'PostgresOperator')
```
Comment (Fokko) [inline tests/contrib/operators/test_sql_operator.py:37]: Can you fix the tests by removing the `skipUnlessImported`? I don't think they are being executed right now.

**Review by Fokko — CHANGES_REQUESTED** [review]
Can you enable the tests? Also there is a MySQL and Postgres available by Docker. Please refer to the other tests as an example.

**Pull request closed without merging**